### DISCUSSION TITLE
Deleted vnet removed from ipam

### DISCUSSION THREAD

**Discussion by janinternet** [Azure/ipam#247] [Q&A] [answered]
We are using 3.0 version Azure ipam. Very happy with it, great tool. We currently see entries removed from the block when vnet resources are deleted in Azure. The reservation gets removed when the vnet is created. But when the vnet is deleted later the entry is removed completely removed from ipam and the cidr get assigned again to new reservation. I've seen discussions here that this could be influenced by a setting. I really want to delete the entry from ipam manually.

**Comment by DCMattyG** [comment]
Hi @janinternet, thank you for reaching out!

Just to clarify, are you asking for the history of the Reservation to be completely removed, or something else?

**Comment by janinternet** [comment]
Thank you for the quick answer. No, not the reservation. We don't want the association to the block to be removed automatically. When we delete the vnet from Azure the cidr is completely removed from ipam and the cidr is made available again. When we delete the vnet we would like to keep this cidr blocked in ipam so it will not get assigned again until we remove it manually.

**Reply by DCMattyG** [reply]
Hi @janinternet, apologies for the delay in response, but I understand now.

The primary reason this isn't the default behavior is that the Azure IPAM product calls out to Azure on a regular basis to check and see what is there and what is not. The biggest concern is perhaps during one of these sync calls, the API for some reason returns incorrect information, or worse returns empty (for whatever reason) and we assume the Virtual Networks are gone. Then we remove them all and it ends up that it was an API call glitch and now your Azure IPAM networks are all screwed up because of an API glitch/blip.

I can certain explore adding an option along the lines of "Auto-remove Virtual Networks after XXX time period" so you could specify like 5 minutes, and hour, or some time frame that if those Virtual Networks aren't detected after that period of time, they are flushed out.

Would that work, or do you need/want the Virtual Network associations to be immediately removed the first time they are no longer detected via API call?

**Reply by janinternet** [reply]
@DCMattyG It would be very nice to have a global setting "Auto-remove Virtual Networks after XXX time period". If people delete resources trough IAC and recreate them later the same  IP space should still be available to them.  If resources are not in use for XXX time they must be removed and made available again. XXX time should be freely configurable. I really would like to set it to 14 days. But larger setups with smaller ip space wanna have this setting smaller.

**Reply by DCMattyG** [reply, marked as answer]
That's a completely valid ask @janinternet. I'll add this to my near-term roadmap and keep you posted once I have something implemented so you can test & provide some feedback.

Thanks for the great suggestion!

**Reply by janinternet** [reply]
@DCMattyG is there any progress?
